client/my-sites/stats/stats-post-detail/index.jsx

```diff
@@ -67,19 +88,44 @@ export default React.createClass( {
 					tabLabel={ this.translate( 'Views' ) } />
 
 				<PostMonths
+					response={ response }
+					isLoading={ isLoading }
 					dataKey="years"
 					title={ this.translate( 'Months and Years' ) }
 					total={ this.translate( 'Total' ) }
-					postViewsList={ this.props.postViewsList } />
+				/>
 
 				<PostMonths
+					response={ response }
+					isLoading={ isLoading }
 					dataKey="averages"
 					title={ this.translate( 'Average per Day' ) }
 					total={ this.translate( 'Overall' ) }
-					postViewsList={ this.props.postViewsList } />
+				/>
 
-				<PostWeeks postViewsList={ this.props.postViewsList } />
+				<PostWeeks response={ response } />
 			</div>
 		);
 	}
 } );
+
+export default connect(
+	function mapStateToProps( state, ownProps ) {
+		const { siteID, postID, statType } = ownProps;
+		const params = {
+			siteID,
+			statType,
+			options: {
```

Comment:
If we pass `postID` here as well, we can automate this in `getStatsItem` & `isStatsItemFetching`.

Ideally, we wrap `connect` in our own util function `statsConnect` (or something) that handles this boilerplate for us in each component.